libraries/AP_Arming/AP_Arming.cpp

```diff
@@ -491,17 +491,27 @@ bool AP_Arming::gps_checks(bool report)
             }
         }
 
-        //GPS OK?
-        if (!AP::ahrs().home_is_set() ||
-            gps.status() < AP_GPS::GPS_OK_FIX_3D) {
-            check_failed(ARMING_CHECK_GPS, report, "Bad GPS Position");
-            return false;
-        }
+        for (uint8_t i = 0; i < gps.num_sensors(); i++) {
+            if (gps.get_type(i) == AP_GPS::GPS_Type::GPS_TYPE_NONE) {
+                if (gps.primary_sensor() == i) {
+                    check_failed(ARMING_CHECK_GPS, report, "GPS primary not configured");
+                    return false;
+                }
+                continue;
+            }
 
-        //GPS update rate acceptable
-        if (!gps.is_healthy()) {
-            check_failed(ARMING_CHECK_GPS, report, "GPS is not healthy");
-            return false;
+            //GPS OK?
+            if (!AP::ahrs().home_is_set() ||
```

Comment:
I do think it is a little odd checking home as part of the GPS checks. Maybe we should move this, although nowhere really jumps out as better.

Reply:
`AP_AHRS::pre_arm_checks` ?

It's gated on the GPS checks being engaged, which is a bit silly, but we can continue to use that in the pre_arm_checks.

Reply:
checking home which is not GPS specific can lead to the wrong GPS being identified, is primary is set to 2nd GPS and no home it will say 1st GPS is bad

Reply:
moved home check to after per-gps checks.